### ISSUE TITLE
Chell's arms appear black at some shader levels

### ISSUE THREAD

**Issue by eriktorbjorn** [ValveSoftware/portal2#255]
I recently played through Portal 2 (single player) for the first time. For the most part, I was blown away at how well it worked (once I had gotten it to run). Good job! However, I did notice a graphics glitch near the very end: In most scenes where you see Chell's arms, they appear to be black and featureless, rather than pink.

This only happens on the "High" and "Very High" shader detail levels. If I turn it down to "Medium" or "Low", it looks fine to me. Se the attached screenshots, which are made at "Very High" and "Medium" levels respectively.

![sp_a4_finale40000](https://cloud.githubusercontent.com/assets/601765/8441199/1b7e6862-1f77-11e5-8c96-4f553b7395e1.jpg)
![sp_a4_finale40001](https://cloud.githubusercontent.com/assets/601765/8441200/1b83896e-1f77-11e5-9047-0f22176a4a7c.jpg)

(This is reminiscent of issue #150, but that is listed a being fixed already. This glitch is much more noticeable in Portal Stories: Mel, since there you always see the hands while carrying the portal gun.)

As you can see from the information below, I'm using the open-source graphics drivers. While they've improved a lot in recent years, I wouldn't really be surprised if this turns out to be a driver bug rather than a Portal 2 bug.

Processor Information:
    Vendor:  GenuineIntel
    CPU Family:  0x6
    CPU Model:  0x3c
    CPU Stepping:  0x3
    CPU Type:  0x0
    Speed:  3400 Mhz
    4 logical processors
    4 physical processors
    HyperThreading:  Unsupported
    FCMOV:  Supported
    SSE2:  Supported
    SSE3:  Supported
    SSSE3:  Supported
    SSE4a:  Unsupported
    SSE41:  Supported
    SSE42:  Supported

Network Information:
    Network Speed:  

Operating System Version:
    Debian GNU/Linux unstable (sid) (64 bit)
    Kernel Name:  Linux
    Kernel Version:  4.0.0-2-amd64
    X Server Vendor:  The X.Org Foundation
    X Server Release:  11701000
    X Window Manager:  Xfwm4
    Steam Runtime Version:  steam-runtime-release_2015-01-06

Video Card:
    Driver:  X.Org Gallium 0.4 on AMD BONAIRE

```
Driver Version:  3.0 Mesa 10.5.8
OpenGL Version: 3.0
Desktop Color Depth: 24 bits per pixel
Monitor Refresh Rate: 60 Hz
VendorID:  0x1002
DeviceID:  0x6658
Number of Monitors:  1
Number of Logical Video Cards:  1
Primary Display Resolution:  1920 x 1080
Desktop Resolution: 1920 x 1080
Primary Display Size: 18,78" x 10,55"  (21,54" diag)
                                        47,7cm x 26,8cm  (54,7cm diag)
Primary VRAM Not Detected
```

Sound card:
    Audio device: Realtek ALC662 rev1

Memory:
    RAM:  7940 Mb

Miscellaneous:
    UI Language:  English
    LANG:  en_DK.utf8
    Microphone:  Not set
    Total Hard Disk Space Available:  234123 Mb
    Largest Free Hard Disk Block:  45518 Mb

Installed software:

Recent Failure Reports:

**Comment by mark101tay** [comment]
This is happening to me and I have Intel HD 3000 on Mesa 10.6.

**Comment by ghost** [comment]
I have this issue, too.
@ Kubuntu 16.04, HD 7950 @ radeonsi, Mesa 11.2.0

**Comment by Tele42** [comment]
Since this issue is trending to be mesa-only, has it also been reported to the mesa devs to see if they have some insight?

**Comment by ghost** [comment]
Hi, good idea. I just opened a bug for it, as I couldn't find a report. Hope I did alright, it's my first report.
https://bugs.freedesktop.org/show_bug.cgi?id=96512

If there's anything more to add, please feel free to comment on the ticket

**Comment by eriktorbjorn** [comment]
For reference, the Mesa bug has been tagged as RESOLVED NOTOURBUG, stating that it

"... uses a shadow sampling instruction on sampler 15, but the texture bound to texture 15 is a 1x1 GL_SRGB texture ... that is, the game is invoking undefined behaviour, hence this is not our bug."

**Comment by mark101tay** [comment]
I will see if turning off specular and/or phong shading will do anything.

**Comment by mark101tay** [comment]
Update: On my computer, turning off phong shading (putting mat_phong 0 in the console) will fix the black arms thing. The same black on Chell's arms covers Chell's whole world model except a little hair, by the way.

**Comment by libcg** [comment]
Could reproduce the bug here too in the ending scene of Portal 2. mesa 12.0.3, R9 Fury.

**Comment by mark101tay** [comment]
I think that it's safe to say that this is a problem for anybody using Mesa. We now need Valve to investigate further into the rendering code and fix it.

**Comment by ghost** [comment]
I agree. With the info from the Mesa bug ticket, I think the code-flaw on Valve's side is a fair assumption. Sadly, there doesn't really seem to be activity from Valve on this so far. 
I also reported this on September 6th in a Steam support ticket and they said they'd pass my info along, so maybe that is already being investigated - and maybe not. It probably would not hurt for someone else to try something similar to make sure. :) This is a long-standing issue and I really hope it gets fixed sooner rather than later.
